### DISCUSSION TITLE
Violation Metrics

### DISCUSSION THREAD

**Discussion by zanebclark** [openmobilityfoundation/curb-data-specification#75] [Data Specification]
@jlarsonOmahaNE brought up a great point in https://github.com/openmobilityfoundation/curb-data-specification/discussions/64. 

>we would want to be able to fine a vehicle for this action but also track how long they were in the no stopping area.

For both [Session](https://github.com/openmobilityfoundation/curb-data-specification/tree/main/metrics#session) and [Aggregate](https://github.com/openmobilityfoundation/curb-data-specification/tree/main/metrics#session) endpoints, we should support compliant metrics along with non-compliant metrics. Non-compliant metrics should include information about how much of the transaction was non-compliant. 

It sounds like this would be useful for prioritizing enforcement resources and re-defining curb policies.

**Comment by jlarsonOmahaNE** [comment]
Are you thinking about adding something to `session_type` as a enforcement/violation tag? Or just use `area` and then add enforcement in `metric_type`?

**Reply by zanebclark** [reply]
I think there's room to combine this with https://github.com/openmobilityfoundation/curb-data-specification/discussions/73. 

If the park wasn't a violation, how long did the park have before it would have been considered one? 
If the park was a violation, by how much? 
What's the violation cost / penalty?

**Reply by jlarsonOmahaNE** [reply]
I think that if it wasn't a violation, it would depend on each city and what they would want to define as time.

For a violation, I think it would be some time past the `max_stay` like 1-5 min?

I think the violation cost would also depend on the rate each city has set. We issue a violation first in a tow zone then call the tow company right away and they have 30 min to get to the vehicle. We can only tow if it is signed Tow Zone. Other areas just get citations such as parking in a bike lane.

**Reply by zanebclark** [reply]
>I think that if it wasn't a violation, it would depend on each city and what they would want to define as time.

Isn't [Methodology](https://github.com/openmobilityfoundation/curb-data-specification/tree/main/metrics#methodology) supposed to be standardized across municipalities? 

>For a violation, I think it would be some time past the max_stay like 1-5 min?

It would compare the event duration to the policy. If the policy had a max_stay of 5 minutes and the vehicle parked for 20 minutes, the overage would be 15 minutes. If the policy had a max_stay of 30 minutes and the vehicle parked for 20 minutes, the overage would be -10 minutes.

**Reply by jlarsonOmahaNE** [reply]
Yes Methodology would standardize it across all cities. So would there be a new metric for enforcement dwell time? And would you need a new `event_type` to label enforcement? See [current](https://github.com/openmobilityfoundation/curb-data-specification/tree/main/events#event-type) event types.

So tracking enforcement time would be something like: `IF event duration > max_stay THEN sum[enforcement time] / count [enforcements]`?

**Reply by zanebclark** [reply]
@jlarsonOmahaNE , I don't think we would need a new `event_type`.  According to a note [here](https://github.com/openmobilityfoundation/curb-data-specification/tree/main/metrics#event-types):

>A "session duration" or "dwell time" can be calculated by calculating the difference between the event_time_start and event_time_end.

I think we would to do two things: 
1. Expand the definition of [Session](https://github.com/openmobilityfoundation/curb-data-specification/tree/main/metrics#session) to include a reference to the policy, or policies, that were "used". Multiple policies couldn't be applied concurrently, but a park duration could extend across multiple policies. I'm not sure the best way to reflect this potentially one-to-many relationship in the `Session` object. For each applicable policy, there are a few metrics that might be useful: 
    * time elapsed before policy was applicable
    * time elapsed when policy was applicable
    * time remaining before the park would have been a violation
    * time elapsed after the park was a violation
    * time elapsed after the policy was applicable
2. Introduce a new [Methodology](https://github.com/openmobilityfoundation/curb-data-specification/tree/main/metrics#methodology) that reflects aggregates for these data points.
